### PR TITLE
feature[TW27313]: Add checks for required roles in Peer Reviews

### PR DESCRIPTION
**Pull request by vaibhavypatel**

### CODE REVIEW

File: plugins/org.eclipse.osee.ats.api/src/org/eclipse/osee/ats/api/review/ReviewRequiredMinimum.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Created on Oct 7, 2024
+ *
```
Comment (dondunne) [inline plugins/org.eclipse.osee.ats.api/src/org/eclipse/osee/ats/api/review/ReviewRequiredMinimum.java:3]: Distro

File: plugins/org.eclipse.osee.ats.api/src/org/eclipse/osee/ats/api/review/ReviewRequiredMinimum.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * PLACE_YOUR_DISTRIBUTION_STATEMENT_RIGHT_HERE
+ */
+package org.eclipse.osee.ats.api.review;
+
+import org.eclipse.osee.ats.api.config.tx.IAtsTeamDefinitionArtifactToken;
+
+public class ReviewRequiredMinimum {
```
Comment (dondunne) [inline plugins/org.eclipse.osee.ats.api/src/org/eclipse/osee/ats/api/review/ReviewRequiredMinimum.java:10]: author

File: plugins/org.eclipse.osee.ats.api/src/org/eclipse/osee/ats/api/review/ReviewRequiredMinimum.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+   private final ReviewRole reviewRole;
+   private final int min;
+   private final IAtsTeamDefinitionArtifactToken parentTeamDef;
+
```
Comment (dondunne) [inline plugins/org.eclipse.osee.ats.api/src/org/eclipse/osee/ats/api/review/ReviewRequiredMinimum.java:15]: Consider adding a test for this feature

**Review by jadenpuckett — APPROVED** [review]

File: plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/workflow/review/role/UserRoleManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      IAtsTeamWorkflow teamWf = DemoUtil.getSawCodeCommittedWf();
+      Collection<IAtsAbstractReview> reviews = AtsApiService.get().getReviewService().getReviews(teamWf);
+      IAtsPeerToPeerReview rev1 = null;
+      for (IAtsAbstractReview revArt : reviews) {
+         if (revArt.getName().contains("algorithm") && revArt instanceof IAtsPeerToPeerReview) {
+            rev1 = (IAtsPeerToPeerReview) revArt;
+         }
+      }
+
+      if (rev1 != null) {
```
Comment (dondunne) [inline plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/workflow/review/role/UserRoleManagerTest.java:52]: You should assert that rev1 not be null.  test should fail if it's not able to setup, not just avoid testing

File: plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/workflow/review/role/UserRoleManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+         ReviewRole TestRole1 = new ReviewRole(1L, "Test Role 1", ReviewRoleType.Reviewer);
+         ReviewRole TestRole2 = new ReviewRole(2L, "Test Role 2", ReviewRoleType.Reviewer);
+         List<ReviewRequiredMinimum> reviewRequiredMinimums1 = new ArrayList<>();
+         reviewRequiredMinimums1.add(
+            new ReviewRequiredMinimum(TestRole1, 1, DemoArtifactToken.SAW_PL_Requirements_TeamDef));
+         List<ReviewRequiredMinimum> reviewRequiredMinimums2 = new ArrayList<>();
+         reviewRequiredMinimums2.add(new ReviewRequiredMinimum(TestRole2, 1, DemoArtifactToken.SAW_Code));
+         List<ReviewRequiredMinimum> reviewRequiredMinimums3 = new ArrayList<>();
+
```
Comment (dondunne) [inline plugins/org.eclipse.osee.ats.ide.integration.tests/src/org/eclipse/osee/ats/ide/integration/tests/ats/workflow/review/role/UserRoleManagerTest.java:61]: Need help understanding this or maybe some javadoc.  Setup one test case, test then go to next one.  This sets up 3 test cases(s)? and then tests?

**Review by dondunne — APPROVED** [review]

**Pull request merged**